heron/instance/src/java/com/twitter/heron/instance/IInstance.java

```diff
@@ -30,7 +31,7 @@
   /**
    * Do the basic setup for HeronInstance
    */
-  void start();
+  void start(State<byte[], byte[]> state);
```

Comment:
Why not use generics in the interface here so instances can handle any types of state?